ring/func/base.py

```diff
@@ -104,27 +104,31 @@ def coerce_function(t):
     # but NEVER add a general iterator processing. it will cause user bugs.
 
 
-def coerce(v):
+def coerce(v, in_memory_storage):
     """Transform the given value to cache-friendly string data."""
+
     type_coerce = coerce_function(type(v))
     if type_coerce:
         return type_coerce(v)
 
     if hasattr(v, '__ring_key__'):
         return v.__ring_key__()
 
+    if in_memory_storage and type(v).__hash__ != object.__hash__:
+        return hash(v)
+
     cls = v.__class__
     if cls.__str__ != object.__str__:
         return str(v)
 
     raise TypeError(
         "The given value '{}' of type '{}' is not a key-compatible type. "
-        "Add __ring_key__() or __str__().".format(v, cls))
+        "Add __ring_key__() or __str__(). __hash__() is also possible in case of in-memory storage.".format(v, cls))
```

Comment:
I think __hash__() related massage should be appended only when `in_memory_storage` is True